lib/models/MatrixAction.js

```diff
@@ -64,10 +64,18 @@ MatrixAction.prototype.formatMentions = function(nickUserIdMap) {
             this.htmlText = this.text;
         }
         userId = ircFormatting.escapeHtmlChars(userId);
+        /* Due to how Riot and friends do push notifications, we need the plain text to match something.
+           Modern client's will pill-ify and will show a displayname, so we can use the localpart so it
+           matches  and doesn't degrade too badly on limited clients.*/
+        const localpart = userId.substr(1, userId.indexOf(":")-1);
```

Comment:
we should be using the display name, as per https://github.com/matrix-org/matrix-doc/pull/1547

Reply:
You *really* want to do a displayname lookup for every match? The localparts already match.

Reply:
I'd prefer it if the cached display name was used, given we already know about the member event.